dev/src/write-batch.ts

```diff
@@ -506,7 +534,22 @@ export class WriteBatch {
    * });
    */
   commit(): Promise<WriteResult[]> {
-    return this.commit_();
+    return this.commit_() as Promise<WriteResult[]>;
+  }
+
+  /**
+   * Commits all pending operations to the database and verifies all
+   * preconditions. Fails the entire write if any precondition is not met.
```

Comment:
Is the latter still true? I would expect that the backend would not be able to enforce this any more.

Reply:
I copied this from the commit() comment and I'm interpreting this to mean that if the the precondition of a single write isn't met the entire batch write fails, which seems to be the case. 
```
await ref.set({foo: 'c'});
await batch.update(ref, {foo: 'a'})
   .delete(ref, { *some invalid update time*});
```
The batch will fail with the error: `Error: 3 INVALID_ARGUMENT: Cannot collapse mutations with different base versions`, and `foo` will still be 'c'.

Reply:
Yeah, but this shouldn't be true for BulkWriter.

Reply:
Ah makes sense. Removed the sentence about preconditions